test/FDTests.jl

```diff
@@ -1742,37 +1743,65 @@ end
     @test isapprox(mat, [1 0; 0 1])
     fn2 = FD.make_function(A, [x, y], in_place=true, init_with_zeros=false)
     mat = [10 10; 10 10]
-    println(mat)
     fn2(mat, [1, 1])
     @test isapprox(mat, [1 10; 10 1])
 
-    #NOT a test because of difficulty and fragility of parsing generated code. You have to verify these by looking at the output.
-    p = make_variables(:p, 21)
-
-    println("NO array zero statement")
-    show(make_Expr(p, p, true, true))
-    show(make_Expr(p, p, true, false))
-    show(make_Expr(p, p, false, true))
-    show(make_Expr(p, p, false, false))
-
-    p[21] = 0
-
-    println("shouldn't have an array zero statement but it should have a p[21]= 0 statement")
-    show(make_Expr(p, p, true, true))
-    println("this should not have an array zero statement nor should have a p[21] = 0 statement")
-    show(make_Expr(p, p, true, false))
-    println("should not have an array zero statement but should have a p[21] = 0 statement")
-    show(make_Expr(p, p, false, true))
-    show(make_Expr(p, p, false, false))
-
-    p[20] = 0
-    println("this should have an array zero statement should not have p[20]=0 or p[21]=0 statementt")
-    show(make_Expr(p, p, true, true))
-    println("this should not have an array zero statement should not have p[20]=0 or p[21]=0 statement")
-    show(make_Expr(p, p, true, false))
-    println("these should both have an array zero creation but should not have p[20]=0 or p[21]=0 statement")
-    show(make_Expr(p, p, false, true))
-    show(make_Expr(p, p, false, false))
+    function check_expected_statements(p, in_place, init_with_zeros)
+        ex = make_Expr(p, p, in_place, init_with_zeros)
+        statements = extract_statements(ex)
+
+        element_type = :(result_element_type = promote_type($(Float64), eltype(input_variables)))
```

Comment:
I understand the motivation for wanting these tests, since there is currently nothing. But my experience is that tests that rely on a particular structure in generated code are problematic. 

A test shouldn't fail if the code being tested is correct. But this code is highly likely to fail if make_Expr() is changed even though the new source may be correct. 

This makes it more difficult to maintain the code. You make a valid change which doesn't affect code correctness but does change :(result .= zero(result_element_type)) to :(res = zero(result_element_type)) and one or more tests fail. Small, correct, changes to the internal layout of generated code then require test code to be rewritten as well. 

If you have enough tests like this soon you don't want to update anything because you know your one line code fix might require patching dozens of lines of test code. How do I know this? There are other tests like this that rely on the internal layout of runtime data structures and they are a pain to maintain.

Is there another way to perform these test that wouldn't depend on the detailed internals of the statements being created by make_Expr()?